ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -2265,6 +2268,50 @@ public static Boolean isToInsertOnlyTable(Table tbl, Map<String, String> props)
     return true;
   }
 
+  public static Boolean isToFullAcid(Table table, Map<String, String> props) {
+    String transactional = props.get(hive_metastoreConstants.TABLE_IS_TRANSACTIONAL);
+    String transactionalProp = props.get(hive_metastoreConstants.TABLE_TRANSACTIONAL_PROPERTIES);
+
+    if (transactional == null && transactionalProp == null) {
+      // Not affected or the op is not about transactional.
+      return false;
+    } else if (transactional == null && table != null) {
+      transactional = table.getParameters().get(hive_metastoreConstants.TABLE_IS_TRANSACTIONAL);
+    }
+
+    if (transactionalProp == null) {
+      boolean isSetToTxn = "true".equalsIgnoreCase(transactional);
+      if (isSetToTxn || table == null) return false; // Assume the full ACID table.
+      throw new RuntimeException("Cannot change '" + hive_metastoreConstants.TABLE_IS_TRANSACTIONAL
+              + "' without '" + hive_metastoreConstants.TABLE_TRANSACTIONAL_PROPERTIES + "'");
+    }
+
+    StorageDescriptor sd = table.getSd();
+    return isStorageDescriptorOrc(sd, table.getTableName());
+  }
+
+
+  private static boolean isStorageDescriptorOrc(StorageDescriptor sd, String tblName) {
```

Comment:
was it copied from UpgradeTool.canBeMadeAcid()? can we extract this to some common place to avoid code duplication?

Reply:
Moved it to AcidUtils.